engine/src/main/java/com/redhat/cloud/notifications/db/repositories/EndpointRepository.java

```diff
@@ -45,106 +40,99 @@
+                .setParameter("endpointType", EMAIL_SUBSCRIPTION)
+                .getResultList();
+        loadProperties(emailEndpoints);
+
+        EmailSubscriptionProperties properties = new EmailSubscriptionProperties();
+        Optional<Endpoint> endpointOptional = emailEndpoints
+                .stream()
+                .filter(endpoint -> properties.hasSameProperties(endpoint.getProperties(EmailSubscriptionProperties.class)))
+                .findFirst();
+        if (endpointOptional.isPresent()) {
+            return endpointOptional.get();
+        }
+
+        Endpoint endpoint = new Endpoint();
+        endpoint.setProperties(properties);
+        endpoint.setAccountId(accountId);
+        endpoint.setEnabled(true);
+        endpoint.setDescription("System email endpoint");
+        endpoint.setName("Email endpoint");
+        endpoint.setType(EMAIL_SUBSCRIPTION);
+        endpoint.prePersist();
+        properties.setEndpoint(endpoint);
+
+        statelessSessionFactory.getCurrentSession().insert(endpoint);
+        statelessSessionFactory.getCurrentSession().insert(endpoint.getProperties());
+        return endpoint;
     }
 
-    public Uni<List<Endpoint>> getTargetEndpoints(String tenant, EventType eventType) {
+    public List<Endpoint> getTargetEndpoints(String tenant, EventType eventType) {
         String query = "SELECT DISTINCT e FROM Endpoint e JOIN e.behaviorGroupActions bga JOIN bga.behaviorGroup.behaviors b " +
                 "WHERE e.enabled = TRUE AND b.eventType = :eventType AND (bga.behaviorGroup.accountId = :accountId OR bga.behaviorGroup.accountId IS NULL)";
 
-        return sessionFactory.withStatelessSession(statelessSession -> {
-            return statelessSession.createQuery(query, Endpoint.class)
-                    .setParameter("eventType", eventType)
-                    .setParameter("accountId", tenant)
-                    .getResultList()
-                    .call(this::loadProperties)
-                    .invoke(endpoints -> {
-                        for (Endpoint endpoint : endpoints) {
-                            if (endpoint.getAccountId() == null) {
-                                if (endpoint.getType() == EMAIL_SUBSCRIPTION) {
-                                    endpoint.setAccountId(tenant);
-                                } else {
-                                    LOGGER.warnf("Invalid endpoint configured in default behavior group: %s", endpoint.getId());
-                                }
-                            }
-                        }
-                    });
-        });
+        List<Endpoint> endpoints = statelessSessionFactory.getCurrentSession().createQuery(query, Endpoint.class)
+                .setParameter("eventType", eventType)
+                .setParameter("accountId", tenant)
+                .getResultList();
+        loadProperties(endpoints);
+        for (Endpoint endpoint : endpoints) {
+            if (endpoint.getAccountId() == null) {
```

Comment:
the null check looks like a check for a possible coding issue, but is actually a business check. Wrapping it in a separate method would make this more clear for others.

Reply:
This PR is only focused on dropping the reactiveness. I'm not changing old code unless it has a direct impact on the reactiveness drop.